lib/javascripts/base_app.js

```diff
@@ -157,8 +157,16 @@ BaseApp.prototype = {
   },
 
   // https://developer.zendesk.com/apps/docs/agent/promises
-  when: function() {
-    return $.when.apply($, arguments);
+  when: function(...promises) {
+    return $.when(...promises.map(promise => {
+      // convert native promises to jQuery promises. See https://stackoverflow.com/a/36255846
+      if (_.isObject(promise) && !$.isFunction(promise.promise) && $.isFunction(promise.then)) {
```

Comment:
Just because this [what jQuery uses](https://j11y.io/jquery/#v=1.7.2&fn=jQuery.when) to check for promises, it probably doesn't matter too much though.